source/blueprints/banking.rst

```diff
@@ -0,0 +1,531 @@
+Banking Assistant
+=================
+
+In this step-by-step walkthrough, you will build a conversational application that allows users to complete common banking tasks that include transferring money and paying bills.
+
+Working through this blueprint will teach you how to
+
+   - obtain missing information (entities) using a slot/entity filling form
+   - query and update external data storage with the dialogue manager
+   - pass immutable information, like a user token, from a client application to your MindMeld application server
+   - use roles in entity hierarchy
+
+.. note::
+
+   Please make sure to install and run all of the :ref:`pre-requisites <getting_started_virtualenv_setup>` for MindMeld before continuing on with this blueprint tutorial.
+
+
+1. The Use Case
+^^^^^^^^^^^^^^^
+
+Users should be able to securely access their banking information and complete tasks as if they are conversing with a teller. They should be able to check balances, pay off credit card debt, transfer money, and carry out various other banking operations they may perform on a day to day basis. In a production application, the frontend would likely handle the secure user authentication and pass a token to the MindMeld server, which would allow the application to make calls to a bank's REST APIs.
+
+2. Example Dialogue Interactions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The conversational flows for a banking assistant can be highly complex, depending on the desired application functionality and the amount of user guidance required at each step. Anticipating all user interactions requires multiple iterations. 
+
+While a real production application will securely authenticate users, for this blueprint, we have given the option to choose from three sample users. You can specify the desired username as a parameter in the request context and if a user is not specified a random user from the sample users will be chosen.
+
+
+Sample usernames: ``splashbro30``, ``larry_l24``, ``johndoe123``
+
+.. code-block:: console
+
+  conv = Conversation(nlp=nlp, app_path='banking_assistant', context={'user_name' : 'splashbro30'})
```

Comment:
Can we change the username to something more conventional, like `Alice` or `Bob`?